theano/sandbox/cuda/cula.py

```diff
@@ -0,0 +1,138 @@
+import theano
+from theano.sandbox.cuda.type import CudaNdarrayType
+from theano.sandbox.cuda import GpuOp
+
+from theano.sandbox.cuda.basic_ops import as_cuda_ndarray_variable
+from theano.sandbox.cuda import cuda_ndarray
+
+cula_available = False
+dimshuffle = cuda_ndarray.cuda_ndarray.dimshuffle
+
+try:
+    from scikits.cuda import cula
+    cula_available = True
+except (ImportError, OSError):
+    pass
+
+cula_initialized = False
+
+
+class GpuSolve(GpuOp):
+    """
+    CULA GPU solver OP.
+
+    trans: Whether to take the transpose of the input matrix
+    or not.
+    """
+    def __init__(self, trans='N'):
+        self.trans = trans
+        super(GpuSolve, self).__init__()
+
+    def __eq__(self, other):
+        return (type(other) == type(self))
+
+    def __hash__(self):
+        return hash(type(self))
```

Comment:
This should use \__props__.  Also 'trans' influences the computation and should be compared.